### PR TITLE
UI fixes

### PR DESCRIPTION
**Pull request by rik86future**

### CODE REVIEW

File: app/controllers/assessments_controller.rb [hunk 1]
```diff
@@ -29,7 +29,7 @@ def set_contact
   end
 
   def log_assessment
-    @need = Need.new(assessment_params.merge(contact_id: @contact.id, name: 'Logged assessment'))
+    @need = Need.new(assessment_params.merge(contact_id: @contact.id, name: 'Logged assessment', start_on: Date.today))
```
Comment (timwis) [inline app/controllers/assessments_controller.rb:32]: Why are we adding a start date? It's not a required field

File: app/models/need.rb [hunk 2]
```diff
@@ -32,11 +34,15 @@ class Need < ApplicationRecord
                    'Other': 'other' }
 
   validates :category, presence: true
+  validates :name, presence: true
+  validate :start_required_for_assessment
 
-  ASSESSMENT_CATEGORIES = ['phone triage', 'check in'].freeze
+  def start_required_for_assessment
```
Comment (timwis) [inline app/models/need.rb:40]: Why is this required?

File: spec/factories/needs.rb [hunk 3]
```diff
@@ -7,7 +7,9 @@
     contact
     category { Need.categories.values.sample }
     name { Faker::Lorem.sentence }
-    start_on { [nil, Faker::Date.between(from: 2.days.ago, to: 6.days.from_now)].sample }
+    start_on do
+      Faker::Date.between(from: 2.days.ago, to: 6.days.from_now) if category.in?(['check in', 'phone triage'])
```
Comment (timwis) [inline spec/factories/needs.rb:11]: Since start_on isn't a required field, shouldn't we keep some null ones in?

File: app/views/shared/_profile-accordion-edit.html.erb [hunk 4]
```diff
@@ -112,7 +115,35 @@
...
+  </div>
+
+  <button class="accordion__control" aria-expanded="true" id="contact-additional-info-accordion">
+    <h2 class="accordion__title">Consents</h2>
+  </button>
+  <div class="accordion__content">
+    <dl class="details-list">
+      <dt class="details-list__caption">Do not call</dt>
+      <dd class="details-list__value" id="no_calls_flag">
+        <%= friendly_boolean @contact.no_calls_flag %>
```
Comment (jamiebuckley) [inline app/views/shared/_profile-accordion-edit.html.erb:130]: Would we also want these to be editable here?

**Comment by timwis** [comment]
Would you mind separating out the UI tweaks from the data validation stuff? We know we want the UI tweaks straight away, but the data validation needs more discussion I think

File: app/views/shared/_profile-accordion-edit.html.erb (hunk 4 above)
Comment (timwis) [reply app/views/shared/_profile-accordion-edit.html.erb:130]: Yes

Comment (timwis) [reply app/views/shared/_profile-accordion-edit.html.erb:130]: From Laura just now:
> when you click on add support action and it opens up the needs and the contact details to edit you should be ablet to add consent there
 but it currently doesn't appear
you can only add it by clicking edit their details
this isn't going to solve the current problem because users won't see it on calls

**Comment by rik86future** [comment]
@timwis and @jamiebuckley 
Regarding this one, I only cherry picked what gone into master and already approved. If I do changes now not only I will create merge conflicts but also the logic will work differently in master and release 2, perhaps we should not include it

here is the original PR: https://github.com/wearefuturegov/beacon/pull/214

If you don't feel like merging it I can close it, but I am not editing it
